delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/TagLabel.scala

```diff
@@ -0,0 +1,46 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.model
+
+import cats.syntax.all._
+import ch.epfl.bluebrain.nexus.delta.sdk.error.FormatError
+import ch.epfl.bluebrain.nexus.delta.sdk.error.FormatError.IllegalTagFormatError
+import io.circe.{Decoder, Encoder}
+
+/**
+  * A safe representation of a tag label
+  *
+  * @param value the string representation of the tag label
+  */
+final case class TagLabel private (value: String) {
+  override def toString: String = value
+}
+
+object TagLabel {
+
+  private[sdk] val maxLength = 32
```

Comment:
Is that the only restriction? I would not like to allow tags with no-ascii characters...that could lead to undesired behaviour. I think I slight modification from Label should actually be enough:
```scala
  private[sdk] val regex = "[a-zA-Z0-9_-[.]]{1,32}".r
```